libs/blocks/icon-block/icon-block.js

```diff
@@ -39,6 +39,13 @@ const iconBlocks = {
   },
 };
 
+//checking if block is part of xx-up and is inline variant
+function upAndInline(el) {
+  const isInlineAndSectionMetadata = el.classList.contains('inline') && el.parentElement?.querySelector('.section-metadata')
+  if (!isInlineAndSectionMetadata) return false;
+  return /(two|three|four|five)[- ]?up/i.test(el.parentElement.querySelector('.section-metadata').textContent);
```

Comment:
nit: missing semicolon and simplifying the test object.
```suggestion
  const isInlineAndSectionMetadata = el.classList.contains('inline') && el.parentElement?.querySelector('.section-metadata');
  if (!isInlineAndSectionMetadata) return false;
  return /(two|three|four|five)[- ]?up/i.test(el.parentElement.className);
  ```

Reply:
Thanks, this looks better.